### PR TITLE
Recognize long and lon for longitude

### PR DESCRIPTION
**Pull request by egrace479**
Added ability to recognize `long` and `lon` for longitude, and a test for the `long` change.

### CODE REVIEW

**Review by hlapp — APPROVED** [review]
👍

**Comment by johnbradley** [comment]
What do you think about ignore case when looking for `lat` and `long`? The Andromeda iNaturalist code creates columns `Lat` and `Long`.

**Comment by egrace479** [comment]
> What do you think about ignoring case when looking for `lat` and `long`? The Andromeda iNaturalist code creates columns `Lat` and `Long`.

It would be simple enough to just set all column names to lowercase with `str.lower` before the check or `str.capitalize` to capitalize them (potentially better for display?).

**Comment by egrace479** [comment]
> > What do you think about ignoring case when looking for `lat` and `long`? The Andromeda iNaturalist code creates columns `Lat` and `Long`.
> 
> It would be simple enough to just set all column names to lowercase with `str.lower` before the check or `str.capitalize` to capitalize them (potentially better for display?).

Realized the capitalization should be carried through as it's likely going to stick with greater generalization of later versions. This made for more files changed, but small changes. @johnbradley do you think that works/makes sense now?

**Review by johnbradley — APPROVED** [review]
LGTM

File: tests/components/test_query.py [hunk 1]
```diff
@@ -48,8 +48,8 @@ def test_get_data(self):
         # Test with mapping = False (no location data)
         df2 = pd.DataFrame(data = {key: data[key] for key in ['Species', 'Subspecies']})
         result_df2, result2_list = get_data(df2, False, features[:2])
-        #self.assertEqual('locality' not in result_df2.columns, True)
-        self.assertEqual(result_df2['locality'].tolist(), ['unknown' for i in range(len(locality))])
+        #self.assertEqual('Locality' not in result_df2.columns, True)
```
Comment (thompsonmj) [inline tests/components/test_query.py:51]: Stray comment to remove?

**Review by thompsonmj — APPROVED** [review]

**Pull request merged**